keras/utils/image_dataset_utils.py

```diff
@@ -384,13 +397,30 @@ def load_image(
     interpolation,
     data_format,
     crop_to_aspect_ratio=False,
+    pad_to_aspect_ratio=False,
 ):
     """Load an image from a path and resize it."""
     img = tf.io.read_file(path)
     img = tf.image.decode_image(
         img, channels=num_channels, expand_animations=False
     )
-    if crop_to_aspect_ratio:
+    
+    if pad_to_aspect_ratio and crop_to_aspect_ratio:
+        raise ValueError(
+                'Only one of pad_to_aspect_ratio, crop_to_aspect_ratio can be set to True'
+            )
+    
+    if pad_to_aspect_ratio:
+        if data_format == "channels_first":
+            img = tf.transpose(img, (2, 0, 1))
```

Comment:
This should be move out of the `if` block to reduce code redundancy.

Also the current logic seems not to support channels_first, please fix: https://github.com/keras-team/keras/actions/runs/8170560185/job/22337254139?pr=19258